### PR TITLE
feature_find_one

### PR DESCRIPTION
**Pull request by tomjoro**
Issue: #29 
  MongoDB: 3.2.0
  Elixir: 1.3.2

This is currently failing tests on my setup: 
   ```
  1) test find_one (Mongo.Test)
     test/mongo_test.exs:127
     Assertion with == failed
     code: nil == Mongo.find_one(c.pid(), coll, %{foo: 43})
     lhs:  nil
     rhs:  0
     stacktrace:
       test/mongo_test.exs:134: (test)
```

### CODE REVIEW

**Comment by tomjoro** [comment]
from @ericmj : 
Enum.at was fixed in Elixir 1.4.0, if you use 1.4.0+ you shouldn't get the bug.

Can we assume version >= 1.4.0 for Elixir?

**Comment by ericmj** [comment]
> Can we assume version >= 1.4.0 for Elixir?

No, keep your current code but add a note that we should use Enum.at when we require 1.4.0+.

File: test/mongo_test.exs [hunk 1]
```diff
@@ -124,6 +124,23 @@ defmodule Mongo.Test do
...
+    assert {:ok, _} = Mongo.insert_one(c.pid, coll, %{foo: 42, bar: 1})
+
+    assert nil == Mongo.find_one(c.pid, coll, %{foo: 43})
+    assert %{"foo" => 42} = Mongo.find_one(c.pid, coll, %{}, batch_size: 1)
+
+    assert {:ok, _} = Mongo.insert_one(c.pid, coll, %{foo: 43})
+
+    assert %{"foo" => 42} = Mongo.find_one(c.pid, coll, %{}, batch_size: 1)
+    # should return the first one so the next test fails
+    assert %{"foo" => 43} != Mongo.find_one(c.pid, coll, %{}, batch_size: 1)
```
Comment (ericmj) [inline test/mongo_test.exs]: I don't think there's a reason to set `:batch_size` here. In fact, maybe we should force it to `1` in the `find_one`function.

File: lib/mongo.ex [hunk 2]
```diff
@@ -364,6 +364,38 @@ defmodule Mongo do
     cursor(conn, coll, query, select, opts)
   end
 
+  @doc """
+  Selects document a single document in a collection and returns either a document
```
Comment (ericmj) [inline lib/mongo.ex]: "Selects a single document".

**Comment by tomjoro** [comment]
This is passing tests now. I think it's ready? Any feedback/changes needed?

**Comment by ericmj** [comment]
See the comments I have left above.

**Comment by ericmj** [comment]
Thank you!

**Pull request merged**